distributed/tests/test_semaphore.py

```diff
@@ -263,26 +268,6 @@ def f(x, sem):
     assert len(s.extensions["semaphores"].leases["x"]) == 1
 
 
-class BrokenComm(Comm):
-    peer_address = ""
-    local_address = ""
-
-    def close(self):
-        pass
-
-    def closed(self):
-        return True
-
-    def abort(self):
-        pass
-
-    def read(self, deserializers=None):
-        raise OSError()
-
-    def write(self, msg, serializers=None, on_error=None):
-        raise OSError()
-
-
 class FlakyConnectionPool(ConnectionPool):
```

Comment:
It would also help to standardize the various fake/flaky/broken connection pools I introduced around the broken comms. I'm with with doing this in another PR, of course